openpype/hosts/houdini/api/lib.py

```diff
@@ -866,6 +866,70 @@ def get_current_context_template_data_with_asset_data():
     return template_data
 
 
+def get_houdini_color_settings():
+    """Get Houdini working file color settings.
+
+    Returns:
+        Dict: The dictionary contains the Houdini working file color settings
+              if the settings are enabled, otherwise it is an empty dictionary.
+    """
+
+    project_settings = get_current_project_settings()
+    color_settings = project_settings["houdini"]["imageio"]["workfile"]
+
+    # Remove leading, and trailing whitespaces
+    color_settings["review_color_space"] = \
+        color_settings["review_color_space"].strip()
+    return color_settings
```

Comment:
I wonder if this function name needs improving. It somewhat sounds like it is getting settings from Houdini itself, but is actually getting a value from the Ayon/OpenPype project settings. However, anything that's clearer does start sounding way too long like `get_houdini_color_project_settings` (even this is ambiguous) or `get_workfile_color_settings_from_project_settings`

What it actually does is _fairly_ limited - I wonder whether it's worth being a library accessible function due to that - I'm fine with it being there though.

Also, there will be many cases where `project_settings` is an already accessible `dict` and doesn't need to be re-queried, so I'm wondering if you should allow `project_settings` to also be an argument to this.

Reply:
> I wonder if this function name needs improving.

Me too. but I have no ideas 😅

>  I wonder whether it's worth being a library accessible function due to that

I added as a library function because I'm using it in two different places. 
so, it's easier to maintain.

https://github.com/ynput/OpenPype/blob/71fe47dc411fa3a0a50fc951b807f868986d7e5f/openpype/hosts/houdini/api/lib.py#L914

https://github.com/ynput/OpenPype/blob/71fe47dc411fa3a0a50fc951b807f868986d7e5f/openpype/hosts/houdini/plugins/publish/validate_review_colorspace.py#L68

> Also, there will be many cases where project_settings is an already accessible dict and doesn't need to be re-queried, so I'm wondering if you should allow project_settings to also be an argument to this.

_doesn't need to be re-queried_,.. what if an admin changed the settings while an artist is opening Houdini 🤔

Reply:
> _doesn't need to be re-queried_,.. what if an admin changed the settings while an artist is opening Houdini 🤔

I meant that e.g. in publishing the Project Settings are already queried once and usually re-used throughout. Similarly project settings are available to Creators, etc. There are many cases where the call might be preferred from something more elaborate that already has a specific project settings queried (which can be a recent query)

Reply:
> In publishing the Project Settings are already queried once and usually re-used throughout.

I got that part. 
but, I can't figure out how to access that piece of data. 
The only way I know is using `apply_settings`: 
The solution would be:
- remove `get_houdini_color_settings()`
- add `apply_settings` in `ValidateReviewColorspace` and `CreateReview`
- pass settings as an argument to `set_review_color_space(some_setting, opengl_node, log=None)`

did I get it right ?

Reply:
I dont think you need `apply_settings` functions for this case because:
1. you dont need the users to sync the settings into the settings existed in either creator/publisher tab.
2. the entire Houdini colorspace implementation doesn't involve any backward compatibility.

Reply:
I'm a little confused.. how could I use Project Settings queried by the publisher ? 

> Excuse me, I have a headache that makes it difficult to concentrate.
> So, please bear with me.

Reply:
It's on the publish context, see [here](https://github.com/ynput/OpenPype/blob/a3e795aa37d99452728e67e7bb2cea0b27f514e9/openpype/plugins/publish/collect_settings.py#L8-L16).

Reply:
> I'm a little confused.. how could I use Project Settings queried by the publisher ?
> 
> > Excuse me, I have a headache that makes it difficult to concentrate.
> > So, please bear with me.
You simply do:
 `project_settings = instance.context.data["project_settings"]`

Reply:
> It's on the publish context, see [here](https://github.com/ynput/OpenPype/blob/a3e795aa37d99452728e67e7bb2cea0b27f514e9/openpype/plugins/publish/collect_settings.py#L8-L16).
> 
> Or is that not what you meant?

> You simply do:
> `project_settings = instance.context.data["project_settings"]`


I can use this in validator but in the creator correct ?

Reply:
Yes, the Creator does not have access to this.

However, on the Creator you have access to `self.project_settings` :)